openlibrary/plugins/upstream/tests/test_account.py

```diff
@@ -21,6 +33,120 @@ def test_create_list_doc(wildcard):
         "expires_on": wildcard
     }
 
+
+def test_process_goodreads_csv():
+    with open_test_data('goodreads_library_export.csv') as reader:
+        csv_data = reader.read()
+
+    # account.process_goodreads_csv() should work with both strings and bytes
+    for d in [csv_data.decode(), csv_data]:
```

Comment:
```suggestion
    for d in [csv_data.decode('utf-8'), csv_data]:
```

Reply:
Does the Py2 test fail on `csv_data.decode()` or on `csv_data`?

Reply:
It looks like the decode() test, since the test failure shows that the value of 'csv' is a unicode string?

```i = <Storage {'csv': u'Book Id [...]```

But it's hard to tell, so I'll just split it up into two tests.

Reply:
After more digging: in Py2, the csv data comes in as a string from ```web.input()```, and in Py3 it comes in as bytes.  The unit test assumed that opening a file with ```mode='rb'``` would mean the result of ```read()``` would be bytes, but in Py2, it's actually a string.  The 'utf-8' encoding you added fixed the first issue identified by the test, but the resulting variable is type 'unicode', not 'str', so the ```i.csv if isinstance(i.csv, str) else i.csv.decode()``` logic results in ```decode()``` being called on a unicode variable, which results in a similar error as trying to decode a string with no encoding:

```
# in Py2
>>> data
'Pippi L\xc3\xa5ngstrump\n'
>>> type(data)
<type 'str'>
>>> data.decode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 7: ordinal not in range(128)
>>> data.decode('utf-8')
u'Pippi L\xe5ngstrump\n'
>>> type(data.decode('utf-8'))
<type 'unicode'>
>>> data.decode('utf-8').decode()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe5' in position 7: ordinal not in range(128)
```

I could adjust the test to just test i.csv in the form it is received (i.e., don't test any form of decode()'d data) , and the Py2 test will test a string, and the Py3 test will test bytes, which covers the two cases I see when testing it myself in my browser.  Thoughts?